model/task/generate_tasks_estimations.go

```diff
@@ -0,0 +1,72 @@
+package task
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/utility"
+	adb "github.com/mongodb/anser/db"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	numRevisionsToSearch = 10
+)
+
+func (t *Task) setGenerateTasksEstimations() error {
+	if !t.GenerateTask || (t.EstimatedNumGeneratedTasks != nil && t.EstimatedNumActivatedGeneratedTasks != nil) {
+		return nil
+	}
+
+	query := db.Query(
+		ByPreviousCommit(
+			t.BuildVariant,
+			t.DisplayName,
+			t.Project,
+			evergreen.RepotrackerVersionRequester,
+			t.RevisionOrderNumber,
```

Comment:
Is the revision order number set for tasks that aren't running for a commit? [This comment](https://github.com/evergreen-ci/evergreen/blob/f0524407c29280cc41d20eeeae67be12d64360b0/model/task/task.go#L194-L195) (and my own recollection) is that the meaning of the revision order number depends on the type of requester. Would be worth double-checking that this query works for different types of versions.

Reply:
updated